### DISCUSSION TITLE
Issue with AzureAD SSO Configuration

### DISCUSSION THREAD

**Discussion by themmini** [netbox-community/netbox#9325] [Q&A] [answered]
Trying to implement Azure AD SSO by following this guide:
https://docs.netbox.dev/en/stable/administration/authentication/microsoft-azure-ad/

I'm able to click the sign in link -> login with AD credentials but I get redirected back to this error message after successfully signing in.

<class 'social_core.exceptions.AuthForbidden'>
Your credentials aren't allowed
Python version: 3.8.1
NetBox version: 3.2.2
![image](https://user-images.githubusercontent.com/11536761/167557690-4f4411eb-2395-4415-9bc8-1a79b0ace963.png)

REMOTE_AUTH_ENABLED = True
REMOTE_AUTH_BACKEND = 'social_core.backends.azuread.AzureADOAuth2'
SOCIAL_AUTH_AZUREAD_OAUTH2_KEY = '{XXXX}'
SOCIAL_AUTH_AZUREAD_OAUTH2_SECRET = '{XXXX}'
SOCIAL_AUTH_REDIRECT_IS_HTTPS = True
REMOTE_AUTH_AUTO_CREATE_USER = True
REMOTE_AUTH_DEFAULT_GROUPS = '{New}'

This is what I put in my config.py - is there anything else I am missing?

**Comment by markkuleinio** [comment, marked as answer]
> SOCIAL_AUTH_AZUREAD_OAUTH2_KEY = '{XXXX}'
SOCIAL_AUTH_AZUREAD_OAUTH2_SECRET = '{XXXX}'

Remove the {}'s from these strings first, restart NetBox and try again.

**Comment by themmini** [comment]
The XXX’s are obscuring the real values
________________________________

**Reply by markkuleinio** [reply]
I'm just saying that if you have the starting { or ending } characters in those strings just like you showed us, it won't work. If you don't have them, that's ok then, keep troubleshooting other things.

**Comment by themmini** [comment]
Ah, my mistake - I’ll try this in the morning.

________________________________

**Comment by themmini** [comment]
Wow that worked thanks so much – now to setup access for users.
I was also looking what else was required to enable guest/external access  - is it just changing the backend source to social_core.backends.azuread_tenant.AzureADTenantOAuth2
?
Or are their additional configuration values required?
